src/controllers/admin/office.ts

```diff
@@ -76,8 +79,94 @@ export const GetAllAdminHandler = async (req: Request, res: Response) => {
       admins: currentAdmins
     });
   } catch (e) {
+    console.log(e)
     return res.status(500).json({
       message: "Failed to retrieve admin details"
     });
   }
 };
+
+export const EditAdminHandler = async (req: Request & { user: { email: string; role: 'Student' | 'Faculty' } }, res: Response) => {
+  try {
+    const validBody = EditAdminValidator.safeParse(req.body);
+    
+    if (!validBody.success) {
+      res.status(400).json({
+        message: "Invalid input data",
+      });
+      throw new CustomError("Invalid input data");
+    }
+
+    return await prismaClient.$transaction(async (tx) => {
+      const adminToEdit = await tx.admin.findFirst({
+        where: {
+          email: req.params.email
```

Comment:
`email` is part of the body and not `params`. I think the understanding of req.params is not being understood clearly here. 

`url/:mySpecialID/details` -> here, `:mySpecialID` is the params from the route. We ought not to put emails within URLs. They are being fetched via the body of the request. 

Also alter the variable name from `adminToEdit` to `existingAdmin` .. this variable will hold the data which has come from the database so this can be further for updation.

Reply:
so i can just use req.body for mail ?
done, replaced all adminToEdit to existingAdmin